Do not trash preserved registers (like 'ebp') when hell node processing

ptomin opened the pull request:
- Use `Platform.CreateTrashedRegisters()` to get trashed register when hell node processing
- Add stack register to list of trashed registers on `win32` and `msdos`
- Output for 7 unit tests was changed. These tests uses `msdos` platform. Some of 86 already broken tests could be affected too, of course.

uxmal commented on src/Decompiler/Analysis/SsaTransform.cs:
```diff
@@ -385,6 +387,14 @@ private void GenerateUseDefsForUnknownCallee(CallInstruction ci)
+                return false;
+            return program.Platform.CreateTrashedRegisters()
+                .Where(r => r.OverlapsWith(stg)).Any();
+        }
```
Calling `Platform.CreateTrashedRegisters()` is only needed once; you could do that outside of this functoin and pass it in as a parameter.

ptomin replied:
Done in 44f49bb

uxmal commented on src/Environments/Windows/Win32Platform.cs:
```diff
@@ -106,6 +106,7 @@ public override HashSet<RegisterStorage> CreateTrashedRegisters()
                 Registers.ecx,
                 Registers.edx,
+                Registers.esp,
             };
```
Don't forget the `Win16Platform`, which is similar to `MsdosPlatform`

ptomin replied:
Done in 85a2f9f

uxmal commented:
Gah! Sorry for the merge conflict; didn't see your PR until after I pushed. I added some comments, please address them and then I'll pull the PR.

ptomin commented:
Never mind! I'll do rebase

ptomin commented:
> I added some comments, please address them

I have passed trashed registers as a parameter of `IsTrashed` and added `sp` to trashed registers list of `win16`. Anything else?

ptomin commented:
Good. Now I'm going to inject `IndirectCallRewriter` to analysis so that it works on `VCExeSample`
